util/git/client.go

```diff
@@ -397,20 +425,122 @@ func (m *nativeGitClient) LsRefs() (*Refs, error) {
 // runs with in-memory storage and is safe to run concurrently, or to be run without a git
 // repository locally cloned.
 func (m *nativeGitClient) LsRemote(revision string) (res string, err error) {
+	set := make(map[SubModule]int)
+	var submoduleRefs []string
+	gitmodules, err := m.loadGitmodulesFile()
```

Comment:
Argo CD executes `LsRemote` before cloning/fetching the repository. This is done this way avoid cloning repository if cache already has manifests required for a given branch/tag. 

So gitmodule file might be missing here, or even worse - we might read gomodule file from a different branch.

Reply:
How would it be possible to read the wrong gitmodules file from a different branch?